src/lsp-server.ts

```diff
@@ -851,7 +851,7 @@ export class LspServer {
         // pending diagnostic requests (regardless of for which file).
         // In general would be better to replace the whole diagnostics handling logic with the one from
         // bufferSyncSupport.ts in VSCode's typescript language features.
-        if (!this.pendingDebouncedRequest && only?.some(kind => kind === CodeActionKind.Source || kind.startsWith(CodeActionKind.Source + '.'))) {
+        if (!this.pendingDebouncedRequest && kinds?.some(kind => TypeScriptAutoFixProvider.kinds.some(k => k.contains(kind)))) {
```

Comment:
This line specifically was not correct as it has matched TS kinds even when the provided kind was `source.foo`, for example.

Reply:
Now fixed.